OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -631,6 +648,132 @@ public async Task<SearchResult<WorkshopCard>> GetNearestByFilter(WorkshopFilter
         return result;
     }
 
+    /// <inheritdoc/>
+    public async Task<SearchResult<WorkshopDto>> FetchByFilterForAdmins(WorkshopFilterAdministration filter = null)
+    {
+        logger.LogInformation("Started retrieving Workshops by filter for admins.");
+
+        if (filter == null)
+        {
+            logger.LogDebug("Method {MethodName} started with null filter. Applying default {Filter}", nameof(FetchByFilterForAdmins), nameof(WorkshopFilterAdministration));
+            filter = new WorkshopFilterAdministration();
+        }
+
+        var predicate = await PredicateBuildForAdminds(filter);
+
+        var workshops = await workshopRepository.Get(
+                skip: filter.From,
+                take: filter.Size,
+                includeProperties: includingPropertiesForMappingDtoModel,
+                whereExpression: predicate,
+                asNoTracking: true)
+            .ToListAsync()
+            .ConfigureAwait(false);
+
+        var workshopsCount = workshops.Count;
+        logger.LogInformation("Retrieved {WorkshopsCount} matching recods by filter for admins.", workshopsCount);
+
+        var workshopsDTO = mapper.Map<List<WorkshopDto>>(workshops);
+
+        var result = new SearchResult<WorkshopDto>()
+        {
+            TotalAmount = workshopsCount,
+            Entities = workshopsDTO,
+        };
+
+        return result;
+    }
+
+    private async Task<Expression<Func<Workshop, bool>>> PredicateBuildForAdminds(WorkshopFilterAdministration filter)
+    {
+        var predicate = PredicateBuilder.True<Workshop>();
+
+        // TODO: add condition for united territorial community admin
+        if (currentUserService.IsMinistryAdmin())
+        {
+            await ApplyFilterForMinistryAdmin(filter).
+                ConfigureAwait(false);
+        }
+        else if (currentUserService.IsRegionAdmin())
+        {
+            var tempPredicate = await ApplyFilterForRegionAdmin(filter).
+                ConfigureAwait(false);
+
+            predicate = predicate.And(tempPredicate);
+        }
+
+        if (!string.IsNullOrEmpty(filter.SearchString))
+        {
+            var tempPredicate = PredicateBuilder.False<Workshop>();
+            var keyWords = filter.SearchString.Split(' ', ',', StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I worry, that `Split` arguments are incorrect. Could you double-check them?

Reply:
The arguments in the Split function correspond to the arguments in other services where filtering by search string is present. As far as I understand, the approach to splitting the string into components before starting the search is the same for all services.

Reply:
Ok, let me double check it for you. From the screenshot below, you see that second parameter in called `Split` function is `int count`. You have no compilation errors, because there is implicit cast between `char` and `int` type. But logic is still bad. 

Please, read what you copy-paste. 
Please, don't trust others code, even your own code. Sometimes even Microsoft devs have problems in their code.
Please, double-check concerned code, especially when it was asked so during code review.
Please, fix your copy of the code, cover it with unit-tests, and create tech-debt item in backlog to fix "other services where filtering by search string is present"

Thanks in advance

![image](https://github.com/user-attachments/assets/a9f41f8f-a5fd-47a1-9338-fc0b3d1a7d53)

Reply:
Sorry, I was thinking about possible punctuation marks for splitting rather than the arguments of the Split function in general. Of course, after your comment, I checked the function's behavior and its possible arguments, and I found the mistake, which I will fix - ```filter.SearchString.Split([' ', ','], StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)```. I tested how the splitting works with the specified arguments and in no case did I ignore this.

Reply:
Thank you, corrected the mistake in the arguments.